data_iterator_sorter.go

```diff
@@ -0,0 +1,92 @@
+package ghostferry
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// DataIteratorSorter is an interface for the DataIterator to choose which order it will process table
+type DataIteratorSorter interface {
+	Sort(unorderedTables map[*TableSchema]uint64) ([]TableMaxPaginationKey, error)
+}
+
+// MaxPaginationKeySorter arranges table based on the MaxPaginationKey in DESC order
+type MaxPaginationKeySorter struct {}
+
+func (s *MaxPaginationKeySorter) Sort(unorderedTables map[*TableSchema]uint64) ([]TableMaxPaginationKey, error) {
+	orderedTables := make([]TableMaxPaginationKey, len(unorderedTables))
+	i := 0
+
+	for k, v := range unorderedTables {
+		orderedTables[i] = TableMaxPaginationKey{k, v}
+		i++
+	}
+
+	sort.Slice(orderedTables, func(i, j int) bool {
+		return orderedTables[i].MaxPaginationKey > orderedTables[j].MaxPaginationKey
+	})
+
+	return orderedTables, nil
+}
+// MaxTableSizeSorter uses `information_schema.tables` to estimate the size of the DB and sorts tables in DESC order
+type MaxTableSizeSorter struct {
+	DataIterator *DataIterator
+}
+
+func (s *MaxTableSizeSorter) Sort(unorderedTables map[*TableSchema]uint64) ([]TableMaxPaginationKey, error) {
+	orderedTables := []TableMaxPaginationKey{}
+	tableNames := []string{}
+	databaseSchemasMap := map[string]int{}
```

Comment:
Minor nitpick, the typical ways of doing a set in golang is:

```suggestion
	databaseSchemasSet := map[string]struct{}{}
```